files-override/app/index.html

```diff
@@ -18,8 +18,8 @@
 
     <script src="/@embroider/core/vendor.js"></script>
     <script type="module">
-      import Application from './app';
-      import environment from './config/environment';
+      import Application from './app/app';
```

Comment:
because `index.html` is in the `app` folder already, the meaning of this import means that `app.js` is actually located at `<repo>/app/app/app.js` -- which is wrong.

I think the change in this PR is papering over a side-effect of the rewritten app still existing.

When observing the output, your change does make sense:
![image](https://github.com/user-attachments/assets/1463f9a1-4f34-4a8e-97d6-5f7169b89922)

However, _in source_, it does not make sense:
![image](https://github.com/user-attachments/assets/f98bbc15-c53e-460a-9938-3c33956048e4)


so the `index.html` would need to move to the root of `files-override` to to have the imports make sense / not be wrong _in source_. 

With the upcoming removal of app rewriting, the proposed change here will make sense _if_ the index.html were moved outside of the app directory.


_However_ (however), the change here also matches what is in the `app-template` base-app in the embroider repo.

We can move the index.html out of `app` later.